src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -406,13 +403,9 @@ private String routePath(AppDeploymentRequest request) {
 				else if (isNotFoundError().test(error)) {
 					logger.warn("Unable to start application. It may have been destroyed before start completed: " + error.getMessage());
 				} else {
-					logger.error(String.format("Failed to start app %s", deploymentId), error);
+					logError(String.format("Failed to start app %s", deploymentId)).accept(error);
 				}
-
-			})
-			.otherwise(isNotFoundError(), e -> {logger.warn("Ignoring spurious 404 error during start: " + e.getMessage()); return Mono.empty();})
```

Comment:
Yep, looks like it's been taken care of by line 406/403 (although message differs).